Assets/SEE/DataModel/DG/GraphSearch.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FuzzySharp;
+
+namespace SEE.DataModel.DG
+{
+    /// <summary>
+    /// Allows searching for nodes by their string representation.
+    /// The graph associated to this search may be dynamic – that is, when the graph changes
+    /// (for example, when a node is added), the search index will be updated accordingly.
+    /// Searches are fuzzy, i.e., they will return results even if the query does not match the
+    /// element's name exactly.
+    ///
+    /// To perform a search on the associated graph, call <see cref="Search"/>.
+    /// </summary>
+    public class GraphSearch : IObserver<ChangeEvent>
+    {
+        /// <summary>
+        /// A mapping from names to a list of nodes with that name.
+        /// Is constructed in the constructor in order not to have to descend into the graph every
+        /// time a search is executed.
+        /// </summary>
+        private readonly IDictionary<string, List<Node>> elements;
+
+        /// <summary>
+        /// The graph to be searched.
+        /// </summary>
+        private readonly Graph graph;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="GraphSearch"/> for the given <paramref name="graph"/>.
+        /// </summary>
+        /// <param name="graph">The graph to be searched.</param>
+        public GraphSearch(Graph graph)
+        {
+            this.graph = graph;
+            elements = graph.Nodes().GroupBy(ElementToString).ToDictionary(g => g.Key, g => g.ToList());
+            graph.Subscribe(this);
+        }
+
+        /// <summary>
+        /// Performs a fuzzy search for the given <paramref name="query"/> in the graph.
+        /// </summary>
+        /// <param name="query">The query to be searched for.</param>
```

Comment:
What does the `query` relate to? `Source.Name`?

Reply:
Yes. The documentation has been updated.